tests/kafkatest/tests/client/downconversion_memory_test.py

```diff
@@ -0,0 +1,119 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from ducktape.mark import parametrize
+from ducktape.mark.resource import cluster
+from ducktape.tests.test import Test
+
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.performance import ProducerPerformanceService, compute_aggregate_throughput, \
+    ConsumerPerformanceService
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.version import DEV_BRANCH, LATEST_0_10, KafkaVersion, LATEST_1_1, LATEST_0_10_0
+
+
+class DownconversionMemoryTest(Test):
+    def __init__(self, test_context):
+        super(DownconversionMemoryTest, self).__init__(test_context=test_context)
+        '''
+        Test Setup:
+        ==========
+        - Java heap size = 190MB
+        - 1M messages, 1kB each ==> 1GB of total messages
+        - Split into 200 partitions ==> approximately 5MB per partition
```

Comment:
Out of curiosity, why does this test need so many partitions? Couldn't we do the same thing with fewer partitions with a larger per-partition limit?

Reply:
I think we could. There is no technical reason behind why I chose so many partitions.